examples/perplexity/perplexity.cpp

```diff
@@ -638,6 +638,9 @@ static results_perplexity perplexity(llama_context * ctx, const gpt_params & par
 
         for (int seq = 0; seq < n_seq_batch; seq++) {
             const float * all_logits = num_batches > 1 ? logits.data() : llama_get_logits_ith(ctx, seq*n_ctx + first);
+            if (!all_logits) {
+                return 1;
```

Comment:
This doesn't match with the return type.

Maybe something like this would be more appropriate?

```suggestion
                return {std::move(tokens), -1, {}, {}};
```